src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -466,6 +471,38 @@ static int open_device(const char* path, bool try_load_moudle = false)
   return nbd;
 }
 
+static bool find_key_from_conf_file(const char *key)
+{
+  std::vector <std::string> my_sections;
+  g_conf->get_my_sections(my_sections);
+  std::string admin_socket_str;
+  return ((g_conf->get_val_from_conf_file(my_sections, key,
+           admin_socket_str, false) == 0) ? true : false);
```

Comment:
I am still not sure about the defaulted asok. In terms of improving ```rbd list```, you can extract the associated pid from "/sys/block/nbdX/pid" and then extract the pool/image from "/proc/PID/cmdline" -- no need for "magic" hard-coded asok paths.